### PR TITLE
kodi: adjust cputemp/gputemp script

### PR DESCRIPTION
**Pull request by lrusak**
Still needs some more work

would be nice if people could post the results of this script
```
#!/bin/bash

for i in /sys/class/hwmon/*; do
  if [ ! -f "$i/name" ]; then
    echo "$(basename $i): $(cat $i/device/name)"
    for j in $i/device/temp*_input; do
      if [ -f "$j" ]; then
        echo "$(basename $j): $(($(cat $j) / 1000)) C"
      fi
    done
  else
    echo "$(basename $i): $(cat $i/name)"
    for j in $i/temp*_input; do
      if [ -f "$j" ]; then
        echo "$(basename $j): $(($(cat $j) / 1000)) C"
      fi
    done
  fi
done
```

### CODE REVIEW

**Comment by MilhouseVH** [comment]
NUC (Skylake i5):
```
hwmon0: acpitz
temp1_input: 27 C
temp2_input: 29 C
hwmon1: coretemp
temp1_input: 40 C <--- Seems to be the same figure reported by `gputemp`?
temp2_input: 39 C
temp3_input: 40 C
hwmon2: iwlwifi
temp1_input: 38 C
```

Revo 3700 (ION2 with 340.102 driver):
```
hwmon0: acpitz
temp1_input: 21 C
temp2_input: 27 C
hwmon1: coretemp
temp2_input: 3 C
temp3_input: 2 C
```

Revo 3700 (ION2 with 340.101 driver):
```
hwmon0: acpitz
temp1_input: 30 C
temp2_input: 31 C
hwmon1: coretemp
temp2_input: 18 C
temp3_input: 17 C
```

Running `gputemp` on the Revo 3700 with 340.101 driver reports `41 C`, and with 340.102 the reported figure is just ` C` (ie. no usable result). It doesn't look like the script in this PR is detecting the nvidia GPU temp.

RPi1/RPi3:
```
hwmon0: bcm2835_thermal
temp1_input: 63 C
```

**Comment by mglae** [comment]
AMD A10-7800 
The **radeon** and **k10temp** values are "correct" AMD virtual temperatures
```
hwmon0: radeon
temp1_input: 8 C
hwmon1: nct6776
temp1_input: 51 C
temp2_input: 41 C
temp3_input: -26 C
temp6_input: 0 C
hwmon2: k10temp
temp1_input: 7 C
hwmon3: fam15h_power
```
AMD C-60
```
hwmon0: acpitz
temp1_input: 68 C
hwmon1: radeon
temp1_input: 67 C
hwmon2: k10temp
temp1_input: 68 C
```

Intel Core i5-650
```
hwmon0: coretemp
temp2_input: 32 C
temp4_input: 34 C
```

Intel Pentium T4200
```
hwmon0: acpitz
temp1_input: 39 C
temp2_input: 38 C
hwmon1: coretemp
temp2_input: 33 C
temp3_input: 34 C
```

**Comment by lrusak** [comment]
Thanks @mglae

**Comment by koenkooi** [comment]
Revo 1600 (ION1), nouveau drivers
```
hwmon0: nouveau
temp1_input: 79 C
```

i3-6100 on a MSI MS-7982/B150M BAZOOKA mainboard
```
hwmon0: acpitz
temp1_input: 27 C
temp2_input: 29 C
hwmon1: coretemp
temp1_input: 57 C
temp2_input: 56 C
temp3_input: 56 C
hwmon2: nct6793
temp10_input: 0 C
temp1_input: 30 C
temp2_input: 127 C
temp3_input: 27 C
temp4_input: 20 C
temp5_input: 127 C
temp6_input: -4 C
temp7_input: 55 C
temp8_input: 0 C
temp9_input: 0 C
```

As expected, my aarch64 boards haven't joined the hwmon movement yet:

S905X (mini m8s II)
```
cat: can't open '/sys/class/hwmon/*/device/name': No such file or directory
*: 
```

Dragonboard-410c
```
cat: '/sys/class/hwmon/*/device/name': No such file or directory
*: 
```

File: packages/mediacenter/kodi/scripts/cputemp [hunk 1]
```diff
@@ -24,51 +24,50 @@
 TEMP=0
 
 if [ $(basename "$0") = "gputemp" -o "$1" = "gpu" ]; then
-  if which lspci >/dev/null; then
+
+  if which lspci 2>/dev/null; then
```
Comment (MilhouseVH) [inline packages/mediacenter/kodi/scripts/cputemp]: Should probably use `command -v lspci >/dev/null`, I think

**Comment by MilhouseVH** [comment]
Looks fine for RPi and Generic - will include in tonight's build and ask people to report any issues with Generic.

File: packages/mediacenter/kodi/scripts/cputemp [hunk 2]
```diff
@@ -24,51 +24,50 @@
...
     if lspci -n | grep 0300 | grep -q 10de; then
       [ -x /usr/bin/nvidia-smi ] && TEMP=`/usr/bin/nvidia-smi -q -x | grep '<gpu_temp>' | awk '{ print $1 }' | sed 's,<gpu_temp>,,g'`
```
Comment (MilhouseVH) [inline packages/mediacenter/kodi/scripts/cputemp]: I guess we can also get rid of the backticks to be consistent with the rest of the script.

Might the following be a slightly more efficient implementation of this logic:
```
  if [ -x /usr/bin/nvidia-smi ]; then
    if command -v lspci >/dev/null && lspci -n | grep 0300 | grep -q 10de; then
      TEMP="$(/usr/bin/nvidia-smi -q -x | awk '/<gpu_temp>/ { print $1 }' | sed 's,<gpu_temp>,,g')"
    fi
  fi
```
It avoids checking for `lspci` when there's no `nvidia-smi` on the system. My thinking is that testing `-x` should be "cheaper" than executing another process such as `which`/`command` - `nvidia-smi` will only ever exist on Nvidia based systems, so a small effciciency gain for everyone else.

Comment (MilhouseVH) [inline packages/mediacenter/kodi/scripts/cputemp]: `grep 0300 | grep -q 10de` is also a bit suspect.

We want to match:
```
01:00.0 0300: 10de:0a64 (rev a2)
```
but could match:
```
01:00.0 10de: beef:0300 (rev a2)
```
or some other variation.

The following is a bit "safer":
```
  if [ -x /usr/bin/nvidia-smi ]; then
    if command -v lspci >/dev/null && lspci -n | grep -q "0300: 10de:"; then
      TEMP="$(/usr/bin/nvidia-smi -q -x | awk '/<gpu_temp>/ { print $1 }' | sed 's,<gpu_temp>,,g')"
    fi
  fi
```

**Comment by MilhouseVH** [comment]
One might even argue that if `-x /usr/bin/nvidia-smi` is true, is there any need to check/call `lspci` at all?

`-x /usr/bin/nvidia-smi` will only ever be true if `xorg-configure` has configured a suitable nvidia video card and thus created the `/var/lib/nvidia-smi` symbolic link that makes `/usr/bin/nvidia-smi`.

**Comment by lrusak** [comment]
@MilhouseVH updated with suggested changes and cleaned up to make the script more uniform.

**Comment by vpeter4** [comment]
Freescale i.MX6 Quad/DualLite (Device Tree)
```
hwmon0: imx_thermal_zone
temp1_input: 49 C
```

**Comment by lrusak** [comment]
squashed

**Pull request merged**